### DISCUSSION TITLE
PorousFlow : Error in Kozeny Carman Equation

### DISCUSSION THREAD

**Discussion by srinath-chakravarthy** [idaholab/moose#21396] [Q&A Modules: General] [answered]
For the kozeny carman equation with option poreperm = kozeny_carman_phi0, the implemented form results in a constant permeability = k0 * k_anisotropy. 

Rationale :
$k_{ij} = A * k_{ij}^0 \frac{\phi^n}{(1-\phi)^m}$
$ A = k0 * (1 - \phi0)^m / \phi0n$
Substituting for $A$ back into the first equation 
$k_{ij} = k_0 * k_{ij}^0$

I checked the implementation and the relevant lines are seen below 
## Calculation of factor A 
`_A = _k0 * std::pow(1.0 - _phi0, _m) / std::pow(_phi0, _n);`

## Permability calc 
` _permeability_qp[_qp] =
      _k_anisotropy * _A * std::pow(_porosity_qp[_qp], _n) / std::pow(1.0 - _porosity_qp[_qp], _m);`

**Comment by WilkAndy** [comment, marked as answer]
Hang on, the `_A` depends on `_phi0` (note the "zero"), while `_permeability_qp[_qp]` depends on `_porosity_qp[_qp]`, which may not be `_phi0`.   I feel like i'm missing something obvious, sorry!

**Reply by srinath-chakravarthy** [reply]
right if the porosity does not change, then _porosity[_qp] is equal to phi0 which means that the permeability will be just one. Also do you have a reference to this equation. Even if there is a porosity change, it seems strange to me, to be multiplying by the inverse of the power term.

**Reply by WilkAndy** [reply]
@hsheldon - can you answer @srinath-chakravarthy 's question about the reference?   @srinath-chakravarthy - the equation seems reasonable to me: as porosity->0 only the numerator survives, and perm = por^n, but as porosity->1, the denominator starts to dominate, and perm = 1/(1-por)^m.  For n and m positive, this seems like the correct type of behaviour.

**Reply by hsheldon** [reply]
@srinath-chakravarthy it looks fine to me; when phi = phi0 you have k = k0 * anisotropy, but when phi != phi0 you have a different value of k (increasing with phi). As phi increases, the numerator increases and the denominator decreases so k increases. Regarding a reference, this equation is essentially the Kozeny-Carman equation; there is a nice explanation of the derivation in this paper (p. 140 in particular): 
Oelkers, E.H., 1996, Physical and chemical properties of rocks and fluids for chemical mass
transport calculations, in Lichtner, P.C., Steefel, C.I. et al., Reactive transport in porous media, Reviews in Mineralogy, v. 34: Washington, Mineralogical Society of America, 131-192.
